src/runtime_src/core/common/api/aie/xrt_graph.cpp

```diff
@@ -204,6 +225,110 @@ send_exception_message(const char* msg)
 
 }
 
+//////////////////////////////////////////////////////////////
+// xrt_graph C++ API implementations (xrt_graph.h)
+//////////////////////////////////////////////////////////////
+namespace xrt {
+
+graph::
+graph(const xrt::device& device, const xrt::uuid& xclbin_id, const std::string& name)
+  : handle(open_graph(device, xclbin_id, name))
+{}
+
+void
+graph::
+reset() const
+{
+  handle->reset();
+}
+
+uint64_t
+graph::
+get_timestamp() const
+{
+  return (handle->get_timestamp());
+}
+
+void
+graph::
+run()
+{
+  handle->run(0);
+}
+
+void
+graph::
+run(uint32_t iterations)
```

Comment:
Put in the header
```suggestion
run(std::uint32_t iterations = 0)
```
and remove the previous overload?

Reply:
Providing two interfaces for run, wait and end, with/without iterations and cycles, are to matching the aietool API so that we have similar concept about run() and run(iterations), etc.

Reply:
I would suggest the same as @keryell. run() can have default of 0.

Reply:
Does it mean we should explicitly remove run() API? It seems we need a clearly definition of 
run(0)
wait(0)
end(0)